functions/Invoke-DbaQuery.ps1

```diff
@@ -372,12 +372,12 @@ function Invoke-DbaQuery {
             Write-Message -Level Debug -Message "SqlInstance passed in, will work on: $instance"
             try {
                 $connDbaInstanceParams = @{
-                    SqlInstance   = $instance
```

Comment:
This change will impact all runs. To minimize impact/change, can you do a test for SqlCredential.UserName match \ or @ and only set it when there's a match?

Reply:
Yes, will do that.

Reply:
After some analysis I have to say: It's not that easy. $instance might be a string, and then I can test SqlCredential.UserName. But: It might by an Azure-AD Account and we would not want a non pooled connection for that, or do we? On the other side $instance might be a Server SMO, and then I would have to test the ConnectionContext.
I also had the idea to do it as in the legacy code path: If a windows credential is used, just force a non pooled connection. But then we have another problem: We need to explicitly close non pooled connections at the end of the command requesting it. But if we don't know if we get a pooled or nonpooled connection, that is impossible and would affect any command.

So now I think that - until we find a better solution and maybe rewrite Invoke-DbaAsync - it is the best way to force a non pooled connection in any case and close it at the end of the command (this is missing now, but I will push that change in a minite).

I'm still open to other solutions, but at the moment I don't see a better one.

Reply:
I think we should focus on the most likely cases. If a SqlCredential is supplied, parse it unless database.microsoft.net (or whatever the name is) is in the SqlInstance name. If someone passes an SMO object, then we don't handle it.

If others disagree, I'm open to that. I'd just like any Invoke-DbaAsync changes to be run past @niphlod

Reply:
Ok, not testing for Azure yet, but for windows credential. Is this the way you would like to go?